docs/migrating.md

```diff
@@ -36,8 +41,79 @@ spec:
       - pause: {}
 ```
 
-## Other Considerations
+!!! warning
+    When migrating a Deployment which is already serving live production traffic, a Rollout should
+    run next to the Deployment before deleting the Deployment or scaling down the Deployment.
+    **Not following this approach might result in downtime**. It also allows for the Rollout to be
+    tested before deleting the original Deployment.
+
+
+## Reference Deployment From Rollout
+
+Instead of removing Deployment you can scale-down in to zero and reference from the Rollout resource:
+
+1. Create a Rollout resource.
+1. Reference an existing Deployment using `workloadRef` field.
+1. Scale-down existing Deployment by changing `replicas` field of an existing Deployment to zero.
+
+Below is an example of a Rollout resource referencing a Deployment.
+
+```yaml
+apiVersion: argoproj.io/v1alpha1               # Create a rollout resource
+kind: Rollout
+metadata:
+  name: rollout-ref-deployment
+spec:
+  replicas: 5
+  workloadRef:                                 # Reference an existing Deployment using workloadRef field
+    apiVersion: apps/v1
+    kind: Deployment
+    name: rollout-ref-deployment
+  strategy:
+    canary:
+      steps:
+        - setWeight: 20
+        - pause: {duration: 10s}
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app.kubernetes.io/instance: rollout-canary
+  name: rollout-ref-deployment
+spec:
+  replicas: 0                                  # Scale down existing deployment
+  selector:
+    matchLabels:
+      app: rollout-ref-deployment
+  template:
+    metadata:
+      labels:
+        app: rollout-ref-deployment
+    spec:
+      containers:
+        - name: rollouts-demo
+          image: argoproj/rollouts-demo:blue
+          imagePullPolicy: Always
+          ports:
+            - containerPort: 8080
+```
+
+Consider following if your Deployment runs in production:
+
+**Running Rollout and Deployment side-by-side**
+
```

Comment:
So, just to completely understand the workflow, the migration will happen like below:
1. Create rollout object to refer to the deployment object.
2. Apply the rollout object, it will create different set of pods with the same spec.
3. Now when all is ready we will scale down the deployment replicas to 0.
4. Then going forward the deployment object can keep to have the pod specs and rollout can keep referring to it and any change in the deployment object will trigger the rollout. And the pods will be managed by rollout as deployment is scaled to 0 (So as in any point of time we can go back to deployment object in control by just changing the replica number).

Please let me know if my understanding is correct.

Reply:
hello @Sonam11, yes. That is correct.